xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-wikimacro/xwiki-platform-rendering-wikimacro-store/src/test/java/org/xwiki/rendering/wikimacro/internal/DefaultWikiMacroTest.java

```diff
@@ -253,10 +253,10 @@ void executeWhenInnerMacro() throws Exception
 
         assertXHTML(
             "<h2 id=\"Hheading\" class=\"wikigeneratedid\"><span>heading</span></h2>"
-                + "<ul class=\"wikitoc\">"
+                + "<ul class=\"wikitoc\"><li class=\"nodirectchild\"><ul>"
                 + "<li><span class=\"wikilink\"><a href=\"#Hheading\">heading</a></span></li>"
-                + "</ul>",
-            "= heading\n\n{{wikimacro1 param1=\"value1\" param2=\"value2\"/}}");
+                + "</ul></li></ul>",
+            "== heading\n\n{{wikimacro1 param1=\"value1\" param2=\"value2\"/}}");
```

Comment:
Why changing what's tested and not the expectation to h1?

Reply:
To provide a good example, with XWIKI-21370m, the goal is to have the page title as the only h1 on the page, which means no h1 in content. 

This is a test, so it's not necessary, but I still think it's good to provide examples of accessible content wherever we can.

Reply:
See https://dev.xwiki.org/xwiki/bin/view/Community/CodeStyle/XWikiXMLFilesCodeStyle/

Reply:
Ok, I don't think there's really a need to change that as it's not wrong (a user could type it), now it's not really what the test was checking so ok to keep it like that.